File: test/language/subtypes/cpp/EnumSubtypeTest.cpp
#include <cstdio>
#include <string>
#include <fstream>

#include "gtest/gtest.h"

#include "subtypes/enum_subtype/Color.h"
#include "subtypes/enum_subtype/CONST_BLACK.h"

namespace subtypes
{
namespace enum_subtype
{

TEST(EnumSubtypeTest, TestSubtype)
{
    ASSERT_EQ(Color::BLACK, CONST_BLACK);
}

} // namespace enum_subtype
} // namespace subtypes

File: test/extensions/language/subtypes/java/subtypes/EnumSubtypeTest.java
package subtypes;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.Test;

import subtypes.enum_subtype.CONST_BLACK;
import subtypes.enum_subtype.Color;

public class EnumSubtypeTest
{
    @Test
    public void testSubtype()
    {
        assertEquals(Color.BLACK, CONST_BLACK.CONST_BLACK);
    }
}
